File: packages/firestore/android/src/main/java/io/capawesome/capacitorjs/plugins/firebase/firestore/classes/options/AddDocumentSnapshotListenerOptions.java
package io.capawesome.capacitorjs.plugins.firebase.firestore.classes.options;

import androidx.annotation.Nullable;

public class AddDocumentSnapshotListenerOptions {

    private String reference;
    private final boolean includeMetadataChanges;

    @Nullable
    private final String serverTimestampBehavior;

    private String callbackId;

    public AddDocumentSnapshotListenerOptions(
        String reference,
        @Nullable Boolean includeMetadataChanges,
        @Nullable String serverTimestampBehavior,
        String callbackId
    ) {
        this.reference = reference;
        this.includeMetadataChanges = includeMetadataChanges == null ? false : includeMetadataChanges;
        this.serverTimestampBehavior = serverTimestampBehavior;
        this.callbackId = callbackId;
    }

    public String getReference() {
        return reference;
    }

    public boolean isIncludeMetadataChanges() {
        return includeMetadataChanges;
    }

    @Nullable
    public String getServerTimestampBehavior() {
        return serverTimestampBehavior;
    }

    public String getCallbackId() {
        return callbackId;
    }
}

File: packages/firestore/ios/Plugin/Classes/Options/AddDocumentSnapshotListenerOptions.swift
import Foundation

@objc public class AddDocumentSnapshotListenerOptions: NSObject {
    private var reference: String
    private var includeMetadataChanges: Bool
    private var serverTimestampBehavior: String?
    private var callbackId: String

    init(reference: String, includeMetadataChanges: Bool, serverTimestampBehavior: String?, callbackId: String) {
        self.reference = reference
        self.includeMetadataChanges = includeMetadataChanges
        self.serverTimestampBehavior = serverTimestampBehavior
        self.callbackId = callbackId
    }

    func getReference() -> String {
        return reference
    }

    func getIncludeMetadataChanges() -> Bool {
        return includeMetadataChanges
    }

    func getServerTimestampBehavior() -> String? {
        return serverTimestampBehavior
    }

    func getCallbackId() -> String {
        return callbackId
    }
}
